src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -562,10 +574,11 @@ impl ThinPool {
 
                 self.thin_pool.set_low_water_mark(get_dm(), lowater)?;
                 self.resume()?;
+                self.set_extend_state(data_extend_failed, meta_extend_failed);
```

Comment:
There is a way out of Failed...  in set_extend_state()

    fn set_extend_state(&mut self, data_extend_failed: bool, meta_extend_failed: bool) {
        let mut new_state = PoolExtendState::Good;

We set it to good if an extension hasn't failed.

I was thinking that each iteration of check() will trigger the same extension attempt, right?  So, if the meta extend failed in one iteration - it will either fail again or something worked and it was extended.

I realize that might not be a good feature of check - that we would keep attempting the extends.

Am I missing something?

Reply:
ah ok, no that is true! haha thanks.